src/components/PokeCard.js

```diff
@@ -44,16 +44,23 @@ const PokeCard = ({
   // get modal content: pokemon species info and moveset
   const triggerModalData = async () => {
     const modalContent = await getPokemonModalAboutContent(dexNo);
-    const { flavor_text_entries, genera } = modalContent;
-    const generaText = findDescriptionByLanguage(genera, "genus", "en");
-    const description = findDescriptionByLanguage(
-      flavor_text_entries,
-      "flavor_text",
-      "en"
-    );
-    setGeneraString(generaText);
-    setFlavorText(description);
-    setModalData(modalContent);
+    // check for undefined modalContent
+    if(modalContent){
+        const { flavor_text_entries, genera } = modalContent;
+        const generaText = findDescriptionByLanguage(genera, "genus", "en");
+        const description = findDescriptionByLanguage(
+          flavor_text_entries,
+          "flavor_text",
+          "en"
+        );
+        setGeneraString(generaText);
+        setFlavorText(description);
+        setModalData(modalContent);
+    } else {
+      setGeneraString('Unavailable');
+      setFlavorText('Unavailable');
+      setModalData({})
+    }
```

Comment:
What does genara stand for?

Reply:
`genera` is an array at the `pokemon-species` endpoint with objects inside. Each object contains `genus`(type: string; but in different foreign languages) and `language` key, with object as value.
<img width="525" alt="Screen Shot 2021-08-12 at 2 40 50 PM" src="https://user-images.githubusercontent.com/16534498/129273850-1d000c4f-6a5b-4c3c-ab34-0016e905447d.png">